test/k8sT/Services.go

```diff
@@ -636,6 +636,35 @@ Secondary Interface %s :: IPv4: (%s, %s), IPv6: (%s, %s)`, helpers.DualStackSupp
 			It("", func() {
 				testNodePort(kubectl, ni, false, false, false, 0)
 			})
+
+			// Explicitly test netfilter compat mode with kube-proxy.
+			SkipContextIf(helpers.DoesNotRunOn54OrLaterKernel, "with reverse NAT at host netfilterCompatMode=true", func() {
+				BeforeAll(func() {
+					DeployCiliumOptionsAndDNS(kubectl, ciliumFilename, map[string]string{
+						"netfilterCompatMode": "true",
+					})
+					ciliumPods, err := kubectl.GetCiliumPods()
+					Expect(err).To(BeNil(), "Cannot get cilium pods")
+					for _, cp := range ciliumPods {
+						err := kubectl.WaitForCiliumPodToCompleteRegeneration(cp, 2*time.Minute)
+						Expect(err).ToNot(HaveOccurred())
+					}
+				})
+
+				AfterAll(func() {
+					DeployCiliumAndDNS(kubectl, ciliumFilename)
```

Comment:
Why is Cilium being redeployed after the test runs?

Reply:
@christarazi To restore the default configuration and ensure subsequent tests are not executed with `netfilterCompatMode=true` I'm guessing.

Reply:
Makes sense, not every test will reconfigure Cilium

Reply:
The main reason to revert `netfilterCompatMode` configuration is to avoid issues like mentioned in this [GH issue](https://github.com/cilium/cilium/issues/17060). Before actual endpoint regeneration completes, we start running the test(s). If the current `bpf host` and previous installation of `bpf host` differs significantly we might run into some issues.